app/models/host/base.rb

```diff
@@ -214,14 +214,18 @@ def set_comment(parser)
       desc 'Note that available facts depend on what facts have been uploaded to Foreman,
            typical sources are Puppet facter, subscription manager etc.
            The facts can be out of date, this macro only provides access to the value stored in the database.'
+      optional :fact_names, Array, desc: 'A list of fact names to return. If empty all facts are returned'
       returns Hash, desc: 'A hash of facts, keys are fact names, values are fact values'
       example '@host.facts # => { "hardwareisa"=>"x86_64", "kernel"=>"Linux", "virtual"=>"physical", ... }', desc: 'Getting all host facts'
       example '@host.facts["uptime"] # => "30 days"', desc: 'Getting specific fact value, +uptime+ in this case'
       aliases :facts
     end
-    def facts_hash
+    def facts_hash(fact_names = [], *args)
+      filtered_names = [fact_names, *args].flatten.compact
       hash = {}
-      fact_values.includes(:fact_name).collect do |fact|
+      query = fact_values.includes(:fact_name)
+      query = query.where(fact_names: {name: fact_names}) if filtered_names.present?
```

Comment:
what's the difference between `fact_names` and `filtered_names`?

Reply:
If I don't include the splatted args, I run into [this sort of thing](https://github.com/theforeman/foreman/pull/9819#discussion_r1320219359) when running the reports.

Reply:
oh wait that `fact_names` should be `filtered_names`, good catch!

Reply:
but you don't pass `filtered_names` to `where()`, you pass `fact_names` which in this case will be just the first thing?

Reply:
(Read the previous two comments in reverse order.)

Updated